File: src/main/java/toker/warbandscripts/panel/entity/ChestSlot.java
package toker.warbandscripts.panel.entity;

import toker.warbandscripts.panel.entity.pk.ChestSlotPK;
import javax.persistence.*;

@Entity
@Table(name = "chest_slot")
@IdClass(ChestSlotPK.class)
public class ChestSlot {

    private Chest chest;
    private Integer slot;
    private Integer ammo;
    private Item item;

    @Id
    @ManyToOne
    @JoinColumns({
            @JoinColumn(name = "chest_index", referencedColumnName = "index"),
            @JoinColumn(name = "chest_server_id", referencedColumnName = "server_id")
    })
    public Chest getChest() {
        return chest;
    }

    public void setChest(Chest chest) {
        this.chest = chest;
    }

    @Id
    @Column(name = "slot")
    public Integer getSlot() {
        return slot;
    }

    public void setSlot(Integer slot) {
        this.slot = slot;
    }

    @ManyToOne
    @JoinColumn(name = "item_id", referencedColumnName = "id", nullable = false)
    public Item getItem() {
        return item;
    }

    public void setItem(Item item) {
        this.item = item;
    }

    @Column(name = "ammo", nullable = false)
    public Integer getAmmo() {
        return ammo;
    }

    public void setAmmo(Integer ammo) {
        this.ammo = ammo;
    }

}

File: src/main/java/toker/panel/entity/ChestSlot.kt
package toker.panel.entity

import toker.panel.entity.pk.ChestSlotPK
import javax.persistence.*

@Entity
@Table(name = "chest_slot")
@IdClass(ChestSlotPK::class)
class ChestSlot(
        @Id
        @ManyToOne
        @JoinColumns(JoinColumn(name = "chest_index", referencedColumnName = "index"), JoinColumn(name = "chest_server_id", referencedColumnName = "server_id"))
        var chest: Chest? = null,

        @Id
        @Column(name = "slot")
        var slot: Int? = null,

        @Column(name = "ammo", nullable = false)
        var ammo: Int? = null,

        @ManyToOne
        @JoinColumn(name = "item_id", referencedColumnName = "id", nullable = false)
        var item: Item? = null
) {
        override fun equals(other: Any?): Boolean {
                if (this === other) return true
                if (javaClass != other?.javaClass) return false
                other as ChestSlot
                if (chest != other.chest) return false
                if (slot != other.slot) return false
                return true
        }

        override fun hashCode(): Int {
                var result = chest?.hashCode() ?: 0
                result = 31 * result + (slot ?: 0)
                return result
        }
}